### ISSUE TITLE
A Gentle Introduction to Statistical Sampling and Resampling

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#465]
A Gentle Introduction to Statistical Sampling and Resampling<br>
<br>
https://ift.tt/2MmONXz<br>
<br>
Jason Brownlee<br>
<br>
<p>Data is the currency of applied machine learning. Therefore, it is important that it is both collected and used effectively.</p>
<p>Data sampling refers to statistical methods for selecting observations from the domain with the objective of estimating a population parameter. Whereas data resampling refers to methods for economically using a collected dataset to improve the estimate of the population parameter and help to quantify the uncertainty of the estimate.</p>
<p>Both data sampling and data resampling are methods that are required in a predictive modeling problem.</p>
<p>In this tutorial, you will discover statistical sampling and statistical resampling methods for gathering and making best use of data.</p>
<p>After completing this tutorial, you will know:</p>
<ul>
<li>Sampling is an active process of gathering observations with the intent of estimating a population variable.</li>
<li>Resampling is a methodology of economically using a data sample to improve the accuracy and quantify the uncertainty of a population parameter.</li>
<li>Resampling methods, in fact, make use of a nested resampling method.</li>
</ul>
<p>Let&rsquo;s get started.</p>
<div class="wp-caption">
<img src="https://machinelearningmastery.com/wp-content/uploads/2018/06/A-Gentle-Introduction-to-Statistical-Sampling-and-Resampling.jpg" alt="A Gentle Introduction to Statistical Sampling and Resampling" width="640" class="wp-image-5413" height="427">
<p class="wp-caption-text">A Gentle Introduction to Statistical Sampling and Resampling<br>
Photo by <a href="https://www.flickr.com/photos/blachswan/33929764716/">Ed Dunens</a>, some rights reserved.</p>
</div>
<h2>Tutorial Overview</h2>
<p>This tutorial is divided into 2 parts; they are:</p>
<ol>
<li>Statistical Sampling</li>
<li>Statistical Resampling</li>
</ol>
<center>
<h3>Need help with Statistics for Machine Learning?</h3>
<p>Take my free 7-day email crash course now (with sample code).</p>
<p>Click to sign-up and also get a free PDF Ebook version of the course.</p>
<p><a href="https://machinelearningmastery.lpages.co/leadbox/142f75173f72a2%3A164f8be4f346dc/5750943224168448/" target="_blank">Download Your FREE Mini-Course</a></p>
</center>
<h2>Statistical Sampling</h2>
<p>Each row of data represents an observation about something in the world.</p>
<p>When working with data, we often do not have access to all possible observations. This could be for many reasons; for example:</p>
<ul>
<li>It may difficult or expensive to make more observations.</li>
<li>It may be challenging to gather all observations together.</li>
<li>More observations are expected to be made in the future.</li>
</ul>
<p>Observations made in a domain represent samples of some broader idealized and unknown population of all possible observations that could be made in the domain. This is a useful conceptualization as we can see the separation and relationship between observations and the idealized population.</p>
<p>We can also see that, even if we intend to use big data infrastructure on all available data, that the data still represents a sample of observations from an idealized population.</p>
<p>Nevertheless, we may wish to estimate properties of the population. We do this by using samples of observations.</p>
<blockquote>
<p>Sampling consists of selecting some part of the population to observe so that one may estimate something about the whole population.</p>
</blockquote>
<p>&mdash; Page 1, <a href="http://amzn.to/2HNgJAQ">Sampling</a>, Third Edition, 2012.</p>
<h3>How to Sample</h3>
<p>Statistical sampling is the process of selecting subsets of examples from a population with the objective of estimating properties of the population.</p>
<p>Sampling is an active process. There is a goal of estimating population properties and control over how the sampling is to occur. This control falls short of influencing the process that generates each observation, such as performing an experiment. As such, sampling as a field sits neatly between pure uncontrolled observation and controlled experimentation.</p>
<blockquote>
<p>Sampling is usually distinguished from the closely related field of experimental design, in that in experiments one deliberately perturbs some part of the population in order to see what the effect of that action is. [&hellip;] Sampling is also usually distinguished from observational studies, in which one has little or no control over how the observations on the population were obtained.</p>
</blockquote>
<p>&mdash; Pages 1-2, <a href="http://amzn.to/2HNgJAQ">Sampling</a>, Third Edition, 2012.</p>
<p>There are many benefits to sampling compared to working with fuller or complete datasets, including reduced cost and greater speed.</p>
<p>In order to perform sampling, it requires that you carefully define your population and the method by which you will select (and possibly reject) observations to be a part of your data sample. This may very well be defined by the population parameters that you wish to estimate using the sample.</p>
<p>Some aspects to consider prior to collecting a data sample include:</p>
<ul>
<li>
<strong>Sample Goal</strong>. The population property that you wish to estimate using the sample.</li>
<li>
<strong>Population</strong>. The scope or domain from which observations could theoretically be made.</li>
<li>
<strong>Selection Criteria</strong>. The methodology that will be used to accept or reject observations in your sample.</li>
<li>
<strong>Sample Size</strong>. The number of observations that will constitute the sample.</li>
</ul>
<blockquote>
<p>Some obvious questions [&hellip;] are how best to obtain the sample and make the observations and, once the sample data are in hand, how best to use them to estimate the characteristics of the whole population. Obtaining the observations involves questions of sample size, how to select the sample, what observational methods to use, and what measurements to record.</p>
</blockquote>
<p>&mdash; Page 1, <a href="http://amzn.to/2HNgJAQ">Sampling</a>, Third Edition, 2012.</p>
<p>Statistical sampling is a large field of study, but in applied machine learning, there may be three types of sampling that you are likely to use: simple random sampling, systematic sampling, and stratified sampling.</p>
<ul>
<li>
<strong>Simple Random Sampling</strong>: Samples are drawn with a uniform probability from the domain.</li>
<li>
<strong>Systematic Sampling</strong>: Samples are drawn using a pre-specified pattern, such as at intervals.</li>
<li>
<strong>Stratified Sampling</strong>: Samples are drawn within pre-specified categories (i.e. strata).</li>
</ul>
<p>Although these are the more common types of sampling that you may encounter, there are other techniques.</p>
<h3>Sampling Errors</h3>
<p>Sampling requires that we make a statistical inference about the population from a small set of observations.</p>
<p>We can generalize properties from the sample to the population. This process of estimation and generalization is much faster than working with all possible observations, but will contain errors. In many cases, we can quantify the uncertainty of our estimates and add errors bars, such as confidence intervals.</p>
<p>There are many ways to introduce errors into your data sample.</p>
<p>Two main types of errors include selection bias and sampling error.</p>
<ul>
<li>
<strong>Selection Bias</strong>. Caused when the method of drawing observations skews the sample in some way.</li>
<li>
<strong>Sampling Error</strong>. Caused due to the random nature of drawing observations skewing the sample in some way.</li>
</ul>
<p>Other types of errors may be present, such as systematic errors in the way observations or measurements are made.</p>
<p>In these cases and more, the statistical properties of the sample may be different from what would be expected in the idealized population, which in turn may impact the properties of the population that are being estimated.</p>
<p>Simple methods, such as reviewing raw observations, summary statistics, and visualizations can help expose simple errors, such as measurement corruption and the over- or underrepresentation of a class of observations.</p>
<p>Nevertheless, care must be taken both when sampling and when drawing conclusions about the population while sampling.</p>
<h2>Statistical Resampling</h2>
<p>Once we have a data sample, it can be used to estimate the population parameter.</p>
<p>The problem is that we only have a single estimate of the population parameter, with little idea of the variability or uncertainty in the estimate.</p>
<p>One way to address this is by estimating the population parameter multiple times from our data sample. This is called resampling.</p>
<p>Statistical resampling methods are procedures that describe how to economically use available data to estimate a population parameter. The result can be both a more accurate estimate of the parameter (such as taking the mean of the estimates) and a quantification of the uncertainty of the estimate (such as adding a confidence interval).</p>
<p>Resampling methods are very easy to use, requiring little mathematical knowledge. They are methods that are easy to understand and implement compared to specialized statistical methods that may require deep technical skill in order to select and interpret.</p>
<blockquote>
<p>The resampling methods [&hellip;] are easy to learn and easy to apply. They require no mathematics beyond introductory high-school algebra, et are applicable in an exceptionally broad range of subject areas.</p>
</blockquote>
<p>&mdash; Page xiii, <a href="http://amzn.to/2G6gMKP">Resampling Methods: A Practical Guide to Data Analysis</a>, 2005.</p>
<p>A downside of the methods is that they can be computationally very expensive, requiring tens, hundreds, or even thousands of resamples in order to develop a robust estimate of the population parameter.</p>
<blockquote>
<p>The key idea is to resample form the original data &mdash; either directly or via a fitted model &mdash; to create replicate datasets, from which the variability of the quantiles of interest can be assessed without long-winded and error-prone analytical calculation. Because this approach involves repeating the original data analysis procedure with many replicate sets of data, these are sometimes called computer-intensive methods.</p>
</blockquote>
<p>&mdash; Page 3, <a href="http://amzn.to/2FVsmVY">Bootstrap Methods and their Application</a>, 1997.</p>
<p>Each new subsample from the original data sample is used to estimate the population parameter. The sample of estimated population parameters can then be considered with statistical tools in order to quantify the expected value and variance, providing measures of the uncertainty of the estimate.</p>
<p>Statistical sampling methods can be used in the selection of a subsample from the original sample.</p>
<p>A key difference is that process must be repeated multiple times. The problem with this is that there will be some relationship between the samples as observations that will be shared across multiple subsamples. This means that the subsamples and the estimated population parameters are not strictly identical and independently distributed. This has implications for statistical tests performed on the sample of estimated population parameters downstream, i.e. paired statistical tests may be required.</p>
<p>Two commonly used resampling methods that you may encounter are k-fold cross-validation and the bootstrap.</p>
<ul>
<li>
<strong>Bootstrap</strong>. Samples are drawn from the dataset with replacement (allowing the same sample to appear more than once in the sample), where those instances not drawn into the data sample may be used for the test set.</li>
<li>
<strong>k-fold Cross-Validation</strong>. A dataset is partitioned into k groups, where each group is given the opportunity of being used as a held out test set leaving the remaining groups as the training set.</li>
</ul>
<p>The k-fold cross-validation method specifically lends itself to use in the evaluation of predictive models that are repeatedly trained on one subset of the data and evaluated on a second held-out subset of the data.</p>
<blockquote>
<p>Generally, resampling techniques for estimating model performance operate similarly: a subset of samples are used to fit a model and the remaining samples are used to estimate the efficacy of the model. This process is repeated multiple times and the results are aggregated and summarized. The differences in techniques usually center around the method in which subsamples are chosen.</p>
</blockquote>
<p>&mdash; Page 69, <a href="http://amzn.to/2Fmrbib">Applied Predictive Modeling</a>, 2013.</p>
<p>The bootstrap method can be used for the same purpose, but is a more general and simpler method intended for estimating a population parameter.</p>
<h2>Extensions</h2>
<p>This section lists some ideas for extending the tutorial that you may wish to explore.</p>
<ul>
<li>List two examples where statistical sampling is required in a machine learning project.</li>
<li>List two examples when statistical resampling is required in a machine learning project.</li>
<li>Find a paper that uses a resampling method that in turn uses a nested statistical sampling method (hint: k-fold cross validation and stratified sampling).</li>
</ul>
<p>If you explore any of these extensions, I&rsquo;d love to know.</p>
<h2>Further Reading</h2>
<p>This section provides more resources on the topic if you are looking to go deeper.</p>
<h3>Books</h3>
<ul>
<li>
<a href="http://amzn.to/2HNgJAQ">Sampling</a>, Third Edition, 2012.</li>
<li>
<a href="http://amzn.to/2FMh8XF">Sampling Techniques</a>, 3rd Edition, 1977.</li>
<li>
<a href="http://amzn.to/2G6gMKP">Resampling Methods: A Practical Guide to Data Analysis</a>, 2005.</li>
<li>
<a href="http://amzn.to/2praHye">An Introduction to the Bootstrap</a>, 1994.</li>
<li>
<a href="http://amzn.to/2FVsmVY">Bootstrap Methods and their Application</a>, 1997.</li>
<li>
<a href="http://amzn.to/2Fmrbib">Applied Predictive Modeling</a>, 2013.</li>
</ul>
<h3>Articles</h3>
<ul>
<li><a href="https://en.wikipedia.org/wiki/Sample_(statistics)">Sample (statistics) on Wikipedia</a></li>
<li><a href="https://en.wikipedia.org/wiki/Simple_random_sample">Simple random sample on Wikipedia</a></li>
<li><a href="https://en.wikipedia.org/wiki/Systematic_sampling">Systematic sampling on Wikipedia</a></li>
<li><a href="https://en.wikipedia.org/wiki/Stratified_sampling">Stratified sampling on Wikipedia</a></li>
<li><a href="https://en.wikipedia.org/wiki/Resampling_(statistics)">Resampling (statistics) on Wikipedia</a></li>
<li><a href="https://en.wikipedia.org/wiki/Bootstrapping_(statistics)">Bootstrapping (statistics) on Wikipedia</a></li>
<li><a href="https://en.wikipedia.org/wiki/Cross-validation_(statistics)">Cross-validation (statistics) on Wikipedia</a></li>
</ul>
<h2>Summary</h2>
<p>In this tutorial, you discovered statistical sampling and statistical resampling methods for gathering and making best use of data.</p>
<p>Specifically, you learned:</p>
<ul>
<li>Sampling is an active process of gathering observations intent on estimating a population variable.</li>
<li>Resampling is a methodology of economically using a data sample to improve the accuracy and quantify the uncertainty of a population parameter.</li>
<li>Resampling methods, in fact, make use of a nested resampling method.</li>
</ul>
<p>Do you have any questions?<br>
Ask your questions in the comments below and I will do my best to answer.</p>
<p>The post <a rel="nofollow" href="https://machinelearningmastery.com/statistical-sampling-and-resampling/">A Gentle Introduction to Statistical Sampling and Resampling</a> appeared first on <a rel="nofollow" href="https://machinelearningmastery.com">Machine Learning Mastery</a>.</p>
<br>
<br>
<br>
<br>
AI|ML|DM|IR|PR<br>
<br>
via Machine Learning Mastery https://ift.tt/2if3vo8<br>
<br>
June 13, 2018 at 03:17AM
